OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -100,9 +103,26 @@ populateFromMarkerData(const TimeSeriesTable_<SimTK::Vec3>& markerTable,
         _markerTable.addTableMetaData("Units", units);
     }
     
+    
+    const auto& allMarkerNamesInFile = markerTable.getColumnLabels();
+
+    // If user specifies a MarkerWeightSet only track markers that it specifies
+    // therefore remove extraneous markers and their data from the Reference
```

Comment:
Can we just change the label by prepending  'do-not-track-me-' and leave the rest of the data table alone?

Reply:
Why is that preferable? It then forces more logic downstream when the number of columns in the underlying container is different from the number of signals. Removal seems cleaner/hassle free to me.

Reply:
The cleanest solution, in my opinion, is to leave the data tables alone and build the logic downstream to filter/mask/weight. If we're going to mess with the Table (under the premise that it's a local copy) then the least expensive option (changing label to anything or even blank) would make more sense, but if there's much logic downstream to force tracking each and every column, I'd leave it as is to minimize change.

Reply:
I don't really follow your rationale. This is the downstream logic.  We are not "messing with the Table"-- the interface takes a const reference to a Table. The fact `MarkersReference` maintains the reference signals internally as a Table is not exposed anywhere to the user. This could be transformed to a Set of splines. This is all implementation detail the user never sees and at this stage trimming the data down to exactly what you need as the Reference signals seems perfectly reasonable/simple.

Reply:
I was advocating for not trimming (which I assumed is more expensive/error prone)  the table just changing the labels but sounds like this didn't turn out to be a major issue in practice, so we'll leave it as is.